test/saml20_clj/sp_test.clj

```diff
@@ -28,3 +29,6 @@
       (is (= (get @mutable 1) nil))
       (is (not= (get @mutable 2) nil)))))
 
+(deftest test-load-large-file
+  (testing "Throws exception when loading large file"
+    (is (thrown? SAXParseException (xml-string->saml-resp (slurp "test-resources/xml-bomb.txt"))))))
```

Comment:
Minor issue, but when the test case fails (e.g. if the size restriction is not honored), the test will throw an NPE error (and will be marked as an Error rather than Fail)

Imho, it would be better to test the `saml20-clj.xml/str->xmldoc` method itself, rather than `xml-string->saml-resp`, as the former does not include SAML-related logic and is only about XML processing

Reply:
The NPE is annoying but a small thing -- I've changed this to test saml20-clj.xml/str->xmldoc